packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2483,6 +2483,18 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertText('Jackson');
   }
 
+  ['@test injecting an unknown service raises an exception']() {
+    this.registerComponent('foo-bar', {
+      ComponentClass: Component.extend({
+        missingService: inject.service()
+      })
+    });
+
+    throws(() => {
```

Comment:
apparently not: https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/container/lib/registry.js#L712

Should this be an assertion?